crates/uv-trampoline/src/bounce.rs

```diff
@@ -94,18 +99,15 @@ fn executable_filename() -> CString {
 
         // That's the error condition because len doesn't include the trailing null byte
         if len as usize == buffer.len() {
-            unsafe {
-                let last_error = GetLastError();
-                match last_error {
-                    ERROR_INSUFFICIENT_BUFFER => {
-                        SetLastError(ERROR_SUCCESS);
-                        // Try again with twice the size
-                        buffer.resize(buffer.len() * 2, 0);
-                    }
-                    err => {
-                        eprintln!("Failed to get executable name: code {}", err);
-                        ExitProcess(1);
-                    }
+            let last_error = unsafe { GetLastError() };
```

Comment:
Nice. I like the reduction in `unsafe` scope. :-)

Reply:
With `std`, we could probably use: https://doc.rust-lang.org/std/io/struct.Error.html#method.last_os_error

And get the error messages that come with that.

Reply:
That would be nice. I would prefer to keep this PR with no_std. I can look into using std separately